Packages/vcs/Lib/VTKPlots.py

```diff
@@ -884,8 +885,64 @@ def plot2D(self,data1,data2,tmpl,gm,vtk_backend_grid=None,vtk_backend_geo=None):
           cot.SetValue(i,levs[i])
         cot.SetValue(Nlevs,levs[-1])
         cot.Update()
-        mapper.SetInputConnection(cot.GetOutputPort())
         mappers = []
+        if gm.label=="y":
+            mapper = vtk.vtkLabeledContourMapper()
+        else:
+            mapper = vtk.vtkPolyDataMapper()
+        lut = vtk.vtkLookupTable()
+        lut.SetNumberOfTableValues(1)
+        lut.SetTableValue(0, 0, 0, 0)
+        if gm.label=="y":
+            mapper.GetPolyDataMapper().SetLookupTable(lut)
+            mapper.SetLabelVisibility(1)
+        else:
+            mapper.SetLookupTable(lut)
+
+        # Create text properties.
+        if gm.label=="y":
+         if gm.text or gm.textcolors:
+          colorOverrides = gm.textcolors if gm.textcolors else [None] * len(gm.text)
+          texts = gm.text if gm.text else [None] * len(gm.textcolors)
+          while len(texts)<Nlevs:
+              texts.append(texts[-1])
+          while len(colorOverrides)<Nlevs:
+              colorOverrides.append(colorOverrides[-1])
+          tprops = vtk.vtkTextPropertyCollection()
+          for tc,colorOverride in zip(texts, colorOverrides):
+              if vcs.queries.istextcombined(tc):
+                  tt,to = tuple(tc.name.split(":::"))
+              elif tc is None:
+                  tt="default"
+                  to="default"
+              elif vcs.queries.istexttable(tc):
+                  tt=tc.name
+                  to="default"
+              elif vcs.queries.istextorientation(tc):
+                  to=tc.name
+                  tt="default"
+              if colorOverride is not None:
+                  tt=vcs.createtexttable(None,tt)
+                  tt.color = colorOverride
+                  tt=tt.name
```

Comment:
nvm, I see now -- the internal tables store it.

We could just pass the override color index to the prepTextProperty method without having to touch the canvas state. I think this would be cleaner, but either way works.